### PR TITLE
Add DOI and ISBNs citations tags

### PR DESCRIPTION
**Pull request by cldambrosio**
This builds on PRs #1527 and #1528 ; adding logic to have `doi` and `isbn` citation tags for Google Scholar indexing:
doi-isbn

@samvera/hyrax-code-reviewers

### CODE REVIEW

File: spec/presenters/hyku/manifest_enabled_work_show_presenter_spec.rb [hunk 1]
```diff
@@ -57,4 +57,29 @@
...
+
+    describe "#isbns" do
+      it "extracts ISBNs from the identifiers" do
+        expect(presenter.isbns).to eq(%w[
+                                        978-83-7659-303-6 978-3-540-49698-4 9790879392788
+                                        3-921099-34-X 3-540-49698-x 0-19-852663-6
+                                      ])
+      end
+    end
+  end
```
Comment (orangewolf) [inline spec/presenters/hyku/manifest_enabled_work_show_presenter_spec.rb]: Can you please add some contexts for when the work doesn't have an identifier and when the identifier doesn't match the two queries?

File: app/presenters/hyku/manifest_enabled_work_show_presenter.rb [hunk 2]
```diff
@@ -35,8 +35,32 @@ def manifest_metadata
...
+      isbn_regex = /(?:ISBN[- ]*13|ISBN[- ]*10|)\s*
+                   ((?:(?:9[\s-]*7[\s-]*[89])?[ -]?(?:[0-9][ -]*){9})[ -]*(?:[0-9xX]))/x
+      isbns = extract_from_identifier(isbn_regex)
+      isbns.flatten unless doi.nil?
+    end
+
     private
 
+      def extract_from_identifier(rgx)
+        unless solr_document['identifier_tesim'].nil?
```
Comment (orangewolf) [inline app/presenters/hyku/manifest_enabled_work_show_presenter.rb]: super minor code style thing, but you might consider "if present?" over "unless nil?".  Double negatives tend to be more confusing.

File: app/presenters/hyku/manifest_enabled_work_show_presenter.rb [hunk 3]
```diff
@@ -35,8 +35,32 @@ def manifest_metadata
...
+      doi = extract_from_identifier(doi_regex)
+      doi.join unless doi.nil?
+    end
+
+    # unlike doi, there can be multiple isbns
+    def isbns
+      isbn_regex = /(?:ISBN[- ]*13|ISBN[- ]*10|)\s*
+                   ((?:(?:9[\s-]*7[\s-]*[89])?[ -]?(?:[0-9][ -]*){9})[ -]*(?:[0-9xX]))/x
+      isbns = extract_from_identifier(isbn_regex)
+      isbns.flatten unless doi.nil?
```
Comment (orangewolf) [inline app/presenters/hyku/manifest_enabled_work_show_presenter.rb]: looks like this should be `unless isbns.nil?` or `if isbns`.  Looks like doi is sneaking in here.

Comment (cldambrosio) [reply app/presenters/hyku/manifest_enabled_work_show_presenter.rb]: Ooops! Thank you.

File: app/presenters/hyku/manifest_enabled_work_show_presenter.rb (hunk 2 above)
Comment (cldambrosio) [reply app/presenters/hyku/manifest_enabled_work_show_presenter.rb]: I agree, will change this thank you.

**Comment by no-reply** [comment]
cc: @vantuyls 
Is this something we'd like to port into Hyrax?

**Comment by cldambrosio** [comment]
Thank you for merging @orangewolf! Sorry there was a conflict - actually we needed to keep both lines to render the `additional_citations` for this to work. Shall I create a new quick PR to add that line in [views/hyrax/base/show](https://github.com/samvera-labs/hyku/pull/1531/commits/b9815f46d41e46577258faf12d358f30b2e02a1d#diff-c901b006e3f6a2a38d2db7e895559cd0) ?

**Comment by orangewolf** [comment]
never fails whenever I think that things are obvious on a conflict.  Next time I'll message you and we can work through it together.  If you make the change on this same branch and just push it again w/ a new pr, I'll get it merged asap

**Comment by cldambrosio** [comment]
:grin: Cool I'm on my way out but will do this tomorrow!

**Pull request merged**